bioimageio/core/resource_io/utils.py

```diff
@@ -268,14 +268,30 @@ def download_uri_to_local_path(uri: typing.Union[raw_nodes.URI, str]) -> pathlib
     return resolve_uri(uri)
 
 
-def _download_uri_to_local_path(uri: raw_nodes.URI) -> pathlib.Path:
+def _download_url_to_local_path(uri: raw_nodes.URI) -> pathlib.Path:
     local_path = BIOIMAGEIO_CACHE_PATH / uri.scheme / uri.authority / uri.path.strip("/") / uri.query
     if local_path.exists():
         warnings.warn(f"found cached {local_path}. Skipping download of {uri}.")
     else:
         local_path.parent.mkdir(parents=True, exist_ok=True)
+
         try:
-            urlretrieve(str(uri), str(local_path))
+            # download with tqdm adapted from:
+            # https://github.com/shaypal5/tqdl/blob/189f7fd07f265d29af796bee28e0893e1396d237/tqdl/core.py
+            # Streaming, so we can iterate over the response.
+            r = requests.get(str(uri), stream=True)
+            # Total size in bytes.
+            total_size = int(r.headers.get("content-length", 0))
+            block_size = 1024  # 1 Kibibyte
+            t = tqdm(total=total_size, unit="iB", unit_scale=True, desc=local_path.name)
+            with local_path.open("wb") as f:
+                for data in r.iter_content(block_size):
+                    t.update(len(data))
+                    f.write(data)
+            t.close()
+            if total_size != 0 and t.n != total_size:
+                # todo: check more carefully and raise on real issue
+                warnings.warn("Download does not have expected size.")
```

Comment:
locally I don't get this warning, but in CI we do: (one commit earlier raises here: https://github.com/bioimage-io/core-bioimage-io-python/runs/4288595508?check_suite_focus=true)
@constantinpape any idea how to raise more carefully here or how to fix the size mismatch in CI?
Otherwise I'd just keep it as a warning for now and not spend too much time on it...

Reply:
Sorry, I don't know what's going on here either. I think we can go ahead with the warning for now and investigate further later.